readme/BREAKING_CHANGES.markdown

```diff
@@ -802,4 +802,94 @@ be made available at
 This change provides the latest features offered by the Portlet 3.0
 Specification, which was released in early 2017.
 
+---------------------------------------
+
+### Changed the Dependency for the liferay-util:html-top JSP tag When Implementing Portlet 3.0
+- **Date:** 2018-Jun-07
+- **JIRA Ticket:** LPS-81983
+
+#### What changed?
+
+When implementing Portlet 3.0, the usage of `portal-kernel`'s `StringBundler`
+has been deprecated in favor of Petra's `StringBundler`.
+
+#### Who is affected?
+
+This affects anyone implementing Portlet 3.0 that uses the
+`<liferay-util:html-top>` JSP tag.
+
+#### How should I update my code?
+
+You must add the following dependency in your build file for your JSPs to
+compile successfully:
+
+**build.gradle**:
+
+    dependencies {
+        ...
+        compileOnly project(":core:petra:petra-string")
+        ...
+    }
+
+**pom.xml**:
+
+    <dependency>
+        <groupId>com.liferay</groupId>
+        <artifactId>com.liferay.petra.string</artifactId>
+        <version>1.2.0</version>
+        <scope>provided</scope>
+    </dependency>
+
+#### Why was this change made?
+
+This change helps provide the latest features offered by the Portlet 3.0
+Specification.
+
+---------------------------------------
+
+### Decoupled Several Classes from PortletURLImpl When Implementing Portlet 3.0
+- **Date:** 2018-Jun-08
+- **JIRA Ticket:** LPS-82119
+
+#### What changed?
+
+All classes implementing `javax.portlet.BaseURL` have been moved to the
+`com.liferay.portlet.internal` sub package. These classes include
+
+- `PortletURLImplWrapper`
+- `LiferayStrutsPortletURLImpl`
+- `StrutsActionPortletURL`
+
+#### Who is affected?
+
+This affects anyone implementing Portlet 3.0 that uses `PortletURLImpl`.
```

Comment:
@codyhoag Change "This affects anyone implementing Portlet 3.0" to "This potentially affects code that attempts to subclass or create a new instance of the  PortletURLImplWrapper, LiferayStrutsPortletURLImpl, or StrutsActionPortletURL." This is typically the case when the `<portlet-url-class>` element is specified in WEB-INF/liferay-portlet.xml -- for more info, see: https://github.com/liferay/liferay-portal/blob/master/definitions/liferay-portlet-app_4_0_0.dtd#L86

Reply:
@ngriffin7a For the `Why was this change made?` section, what do you propose we write? Since it's not for the Portlet 3.0 implementation, I'm not sure what to write there 😄

Reply:
@codyhoag The greek root word behind "petra" means "rock", and so by extracting our stable utilities from portal-kernel into various "petra" modules helps stabilize the foundation of our products. Here is a list: https://search.maven.org/#search%7Cga%7C1%7Cpetra

Reply:
@codyhoag At some point we need to move away from the portal-kernel version of things to the petra versions in order to fully enjoy the stability.

Reply:
@ngriffin7a cool; that explains the Petra breaking change. I'll add that there. What's the rationale behind decoupling classes from `PortletURLImpl`?

Reply:
@codyhoag Implementation classes are just that, implementations of an API. Having anything that relies on an implementation-specific detail within an API is to be avoided. This is something that should have been corrected before 7.0 was released.

Reply:
@ngriffin7a makes sense. Thanks!